### PR TITLE
Make X509_digest,X509_CRL_digest public

### LINKED ISSUE #2124: X509: cert_info no longer accessible
While porting an application to OpenSSL 1.1.0, I found that the `cert_info` struct member that was previously freely accessible for `X509` data types seems now no longer accessible by any means.

If possible, please add a function or macro to access `X509->cert_info`.

The application I am porting hashes this field and uses the hash as part of a certificate representation. If `cert_info` can now no longer be exposed, please suggest a different part of the data I can use for the same purpose.

Thank you!

### PR DESCRIPTION
**Pull request by richsalz**
Fixed #2124, #2125.

### CODE REVIEW

File: doc/man3/X509_digest.pod [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ int X509_pubkey_digest(const X509 *data, const EVP_MD *type,
+                        unsigned char *md, unsigned int *len);
+
+ int X509_REQ_digest(const X509_REQ *data, const EVP_MD *type,
+                     unsigned char *md, unsigned int *len);
+
+ int X509_NAME_digest(const X509_NAME *data, const EVP_MD *type,
+                      unsigned char *md, unsigned int *len);
+
+ int PKCS7_ISSUER_AND_SERIAL_digest(PKCS7_ISSUER_AND_SERIAL *data,
```
Comment (levitte) [inline doc/man3/X509_digest.pod]: Not sure if this function actually belongs here...

File: doc/man3/X509_digest.pod [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                                    const EVP_MD *type, unsigned char *md,
+                                    unsigned int *len);
+
+=head1 DESCRIPTION
+
+X509_pubkey_digest() returns a digest of the DER representation of the public
+key in the specified X509 B<data> object.
+All other functios described here return a digest of the DER represenation
```
Comment (levitte) [inline doc/man3/X509_digest.pod]: functions

Comment (levitte) [inline doc/man3/X509_digest.pod]: representation

File: doc/man3/X509_digest.pod (hunk 1 above)
Comment (levitte) [inline doc/man3/X509_digest.pod]: ... not sure if my thinking is right, but I find it surprising to find a PKCS7 function on a X509 manual page...

File: doc/man3/X509_digest.pod (hunk 2 above)
Comment (richsalz) [inline doc/man3/X509_digest.pod]: fixed

File: doc/man3/X509_digest.pod (hunk 1 above)
Comment (richsalz) [inline doc/man3/X509_digest.pod]: I understand that, but look at BIO_get_ex_new_index.pod  And recall that the manpage is installed under all the filenames.

**Comment by levitte** [comment]
Approval pending build results.

Also, please squash those two commits into one.

**Comment by richsalz** [comment]
squashed and pushed updated commit.  also okay for 1.1.0 ?  since this is caused by opaque work.

**Comment by levitte** [comment]
> also okay for 1.1.0 ?

Yes

**Comment by richsalz** [comment]
3e5d9da on master and 550f0f9 on 1.1.0

**Pull request closed without merging**